backend/__init__.py

```
@@ -14,23 +17,36 @@ def create_app():
+        CORS(app, 
+             origins=lambda: True,  # Allow all origins dynamically
+             supports_credentials=True,  # Enable credentials for session cookies
+             methods=['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'],
+             allow_headers=['Content-Type', 'Authorization', 'X-Requested-With', 'Accept', 'Origin'],
+             expose_headers=['Content-Type', 'Authorization', 'X-Requested-With'],
+             max_age=3600)
+    else:
+        # In production, use configured allowed origins
+        cors_origins = os.environ.get('CORS_ALLOWED_ORIGINS')
+        if not cors_origins:
+            raise ValueError("CORS_ALLOWED_ORIGINS environment variable must be configured for production")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)